src/components/App.tsx

```diff
@@ -4261,6 +4260,41 @@ class App extends React.Component<AppProps, AppState> {
       if (this.state.selectedLinearElement) {
         const linearElementEditor =
           this.state.editingLinearElement || this.state.selectedLinearElement;
+
+        const ret = LinearElementEditor.addMidpoint(
+          this.state.selectedLinearElement,
+          pointerCoords,
+          this.state,
+        );
```

Comment:
`addMidpoint` should be split into `shouldAddMidpoint` and `addMidpoint` methods